[CEP XXXX] Versioning of Existing conda Standards (#119)
conda/ceps: cep-XXXX.md

# CEP XXXX - Versioning of Existing conda Standards

<table>
<tr><td> Title </td><td> Versioning of Existing conda Standards and Specs
<tr><td> Status </td><td> Draft  </td></tr>
<tr><td> Author(s) </td><td> Matthew R. Becker &lt;becker.mr@gmail.com&gt;</td></tr>
<tr><td> Created </td><td> Mar 14, 2025</td></tr>
<tr><td> Updated </td><td> Mar 14, 2025</td></tr>
<tr><td> Discussion </td><td> NA </td></tr>
<tr><td> Implementation </td><td> NA </td></tr>
</table>

## Abstract

As the conda ecosystem is being standardized and evolving, the existing standards
(in many cases only documented by the code that implements them) have been called
`v1` in some cases (e.g., `.tar.bz2` versus `.conda`) and `v0` in others (e.g., `v0` versus `v1` recipe formats).
This differing nomenclature is confusing. In this CEP, we resolve this issue by declaring all existing `conda`
standards, documented formally in CEPs or not, the `v1` standards.

## Specification

All existing conda standards which satisfy the following conditions

- are documented formally in CEPs or are only effectively documented through code implementations

> - have not been superseded by a newer standard labeled `v2`

Author: ```suggestion
- have not been superseded by a newer standard labeled `v1`
```

MUST be versioned at `v1`.

> The numbering of recipe formats is exempt from this CEP.

Author: ```suggestion
The numbering of recipe formats is exempt from this CEP.

Standards MUST use a version specified that matches the regex `^v[0-9]+($|.[0-9]+$)`.

```

# CEP XXXX - Versioning of Existing conda Standards

<table>
<tr><td> Title </td><td> Versioning of Existing conda Standards and Specs
<tr><td> Status </td><td> Draft  </td></tr>
<tr><td> Author(s) </td><td> Matthew R. Becker &lt;becker.mr@gmail.com&gt;</td></tr>
<tr><td> Created </td><td> Mar 14, 2025</td></tr>
<tr><td> Updated </td><td> Mar 14, 2025</td></tr>
<tr><td> Discussion </td><td> NA </td></tr>
<tr><td> Implementation </td><td> NA </td></tr>
</table>

## Abstract

As the conda ecosystem is being standardized and evolving, the existing standards
(in many cases only documented by the code that implements them) have been called
`v1` in some cases (e.g., `.tar.bz2` versus `.conda`) and `v0` in others (e.g., `v0` versus `v1` recipe formats).
This differing nomenclature is confusing. In this CEP, we resolve this issue by declaring all existing `conda`
standards, documented formally in CEPs or not, the `v0` standards.

## Specification

All existing conda standards which satisfy the following conditions

- are documented formally in CEPs or are only effectively documented through code implementations

> - have not been superseded by a newer standard labeled `v1`

Author: ```suggestion
- have not been superseded by a newer standard labeled `v2`
```

> MUST be versioned at `v0`.

Author: ```suggestion
MUST be versioned at `v1`.
```

> The number of conda package formats, `.tar.bz2` and `.conda`, is exempt from this CEP.

Author: ```suggestion
The numbering of conda package formats, `.tar.bz2` and `.conda`, is exempt from this CEP.
```

> The numbering of conda package formats, `.tar.bz2` and `.conda`, is exempt from this CEP.

Author: ```suggestion
The numbering of conda package formats (e.g., `.tar.bz2` and `.conda`) is exempt from this CEP.
```

> The numbering of conda package formats (e.g., `.tar.bz2` and `.conda`) is exempt from this CEP.

Author: ```suggestion
The numbering of recipe formats is exempt from this CEP.
```

## Backwards Compatibility

> The `v0` versioning is fully backwards compatible with all current standards, except conda package formats
> which have been explicitly excluded.

Author: ```suggestion
The `v1` versioning is fully backwards compatible with all current standards, except recipe formats which have been explicitly excluded.
```

> # CEP XXXX - Versioning of Existing Conda Standards

Author: ```suggestion
# CEP XXXX - Versioning of Existing conda Standards
```

<table>

> <tr><td> Title </td><td> Versioning of Existing Conda Standards and Specs

Author: ```suggestion
<tr><td> Title </td><td> Versioning of Existing conda Standards and Specs
```

<tr><td> Status </td><td> Draft  </td></tr>
<tr><td> Author(s) </td><td> Matthew R. Becker &lt;becker.mr@gmail.com&gt;</td></tr>
<tr><td> Created </td><td> Mar 14, 2025</td></tr>
<tr><td> Updated </td><td> Mar 14, 2025</td></tr>
<tr><td> Discussion </td><td> NA </td></tr>
<tr><td> Implementation </td><td> NA </td></tr>
</table>

## Abstract

As the conda ecosystem is being standardized and evolving, the existing standards 
(in many cases only documented by the code that implements them) have been called 
`v1` in some cases (e.g., `.tar.bz2` versus `.conda`) and `v0` in others (e.g., `v0` versus `v1` recipe formats). 
This differing nomenclature is confusing. In this CEP, we resolve this issue by declaring all existing `conda` 

> standards, documented formally in CEPs or not, the `v0` standards. Changes to these standards then are labeled `v1`, 

Author: ```suggestion
standards, documented formally in CEPs or not, the `v0` standards.
```

## Specification

All existing conda standards which satisfy the following conditions

- are documented formally in CEPs or are only effectively documented through code implementations
- have not been superseded by a newer standard labeled `v1`

MUST be versioned at `v1`.

The numbering of build recipe formats is exempt from this CEP.

> Standards MUST use a version specifier that matches the regex `^v?[0-9]+($|.[0-9]+$)`. The `v` MAY be omitted.

Author: ```suggestion
Standards MUST use a version specifier that consists only of numbers and `.`s, has a major version, and has at most a minor version (i.e. matches the regex `^[0-9]+($|\.[0-9]+$)`). A `v` MAY be prepended for clarity, but MUST be removed before comparing versions. Version comparisons and ordering MUST follow conda's version comparison and ordering conventions.
```

# CEP XXXX - Versioning of Existing conda Standards

<table>

> <tr><td> Title </td><td> Versioning of Existing conda Standards and Specs

Reviewer: ```suggestion
<tr><td> Title </td><td> CEP XXXX - Versioning of Existing conda Standards and Specs </td></tr>
```

<tr><td> Status </td><td> Draft  </td></tr>
<tr><td> Author(s) </td><td> Matthew R. Becker &lt;becker.mr@gmail.com&gt;</td></tr>
<tr><td> Created </td><td> Mar 14, 2025</td></tr>
<tr><td> Updated </td><td> Mar 14, 2025</td></tr>
<tr><td> Discussion </td><td> NA </td></tr>
<tr><td> Implementation </td><td> NA </td></tr>
</table>

## Abstract

As the conda ecosystem is being standardized and evolving, the existing standards
(in many cases only documented by the code that implements them) have been called
`v1` in some cases (e.g., `.tar.bz2` versus `.conda`) and `v0` in others (e.g., `v0` versus `v1` recipe formats).
This differing nomenclature is confusing. In this CEP, we resolve this issue by declaring all existing `conda`
standards, documented formally in CEPs or not, the `v1` standards.

## Specification

All existing conda standards which satisfy the following conditions

- are documented formally in CEPs or are only effectively documented through code implementations
- have not been superseded by a newer standard labeled `v1`

MUST be versioned at `v1`.

> The numbering of recipe formats is exempt from this CEP.

Reviewer: ```suggestion
The numbering of build recipe formats is exempt from this CEP.
```

For clarity in case there are other things called "recipe" out there.

<table>
<tr><td> Title </td><td> CEP XXXX - Versioning of Existing conda Standards and Specs </td></tr>
<tr><td> Status </td><td> Draft  </td></tr>
<tr><td> Author(s) </td><td> Matthew R. Becker &lt;becker.mr@gmail.com&gt;</td></tr>
<tr><td> Created </td><td> Mar 14, 2025</td></tr>
<tr><td> Updated </td><td> Mar 14, 2025</td></tr>
<tr><td> Discussion </td><td> NA </td></tr>
<tr><td> Implementation </td><td> NA </td></tr>
</table>

## Abstract

As the conda ecosystem is being standardized and evolving, the existing standards
(in many cases only documented by the code that implements them) have been called
`v1` in some cases (e.g., `.tar.bz2` versus `.conda`) and `v0` in others (e.g., `v0` versus `v1` recipe formats).
This differing nomenclature is confusing. In this CEP, we resolve this issue by establishing conventions on how to number existing and future `conda` standards.

## Specification

All existing conda standards which satisfy the following conditions

- are documented formally in CEPs or are only effectively documented through code implementations
- have not been superseded by a newer standard labeled `v1`

MUST be versioned at `v1`.

> The numbering of build recipe formats is exempt from this CEP.

Reviewer 2: I agree given our past conversations. But do we want to codify the reasoning? I think that saying something along the lines of "we use V0 in the old format because the new format has its schema field set to 1" would be a nice breadcrumb for future folks who'll ask why there's nuance here.

> The numbering of build recipe formats is exempt from this CEP.

Author: ```suggestion
The numbering of build recipe formats is exempt from this CEP due to the fact
that the new recipe format in [CEP 14](https://github.com/conda/ceps/blob/main/cep-0014.md) specified `schema_version: 1`.
```

> Standards MUST use a version specifier that consists only of numbers and `.`s, has a major version,

Author: ```suggestion
Standards MUST use a version specifier that consists only of numbers and a single period `.`, has a major version,
```

# CEP XXXX - Versioning of Existing conda Standards

<table>
<tr><td> Title </td><td> CEP XXXX - Versioning of Existing conda Standards and Specs </td></tr>
<tr><td> Status </td><td> Draft  </td></tr>
<tr><td> Author(s) </td><td> Matthew R. Becker &lt;becker.mr@gmail.com&gt;</td></tr>
<tr><td> Created </td><td> Mar 14, 2025</td></tr>
<tr><td> Updated </td><td> Mar 14, 2025</td></tr>
<tr><td> Discussion </td><td> NA </td></tr>
<tr><td> Implementation </td><td> NA </td></tr>
</table>

## Abstract

As the conda ecosystem is being standardized and evolving, the existing standards
(in many cases only documented by the code that implements them) have been called
`v1` in some cases (e.g., `.tar.bz2` versus `.conda`) and `v0` in others (e.g., `v0` versus `v1` recipe formats).

> This differing nomenclature is confusing. In this CEP, we resolve this issue by declaring all existing `conda`
> standards, documented formally in CEPs or not, the `v1` standards.

Author: ```suggestion
This differing nomenclature is confusing. In this CEP, we resolve this issue by establishing conventions on how to number existing and future `conda` standards.
```
